instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,47 @@
+import React,{Fragment} from 'react';
+import Comment from './Comment'
+import PropTypes from 'prop-types';
+
+class CommentSection extends React.Component {
+    constructor(props){
+        super(props);
+        this.state= {
+            comments: props.comments,
+            input: ''
+        }
+    }
+    handleChange = event => {
+        this.setState({
+            input: event.target.value
+        })
+    }
+    addNewComment = event => {
+        event.preventDefault();
+        // console.log('type',typeof this.state.comments, 'comment', this.state.comments);
+
+       this.setState({ comments: [...this.state.comments, {username:'holybanana', text:this.state.input}],input: ''})
```

Comment:
Nice job finding getting you add new comment working and troubleshooting that error you were having.